### PR TITLE
Simplify how the role can be used from playbooks

### PR DESCRIPTION
**Pull request by sebastian-luna-valero**
With this we are a step closer to https://github.com/ansible-roles/ansible-role-template

### CODE REVIEW

**Comment by sebastian-luna-valero** [comment]
Changelog:
* Moved all `vars` back to default. This way `vars` will be automatically picked up by the playbook and they don't need to be explicitly imported
* The `preps.yml` in `tasks` was not running for worker nodes. With this refactoring, `preps.yml` is now applied.
* Added further checks in the `tasks/main.yml` as first thing about the existence of networked folders for `sessiondir` and `cachedir`. Otherwise, the role will silently create them in the default values.

File: playbook_arc_cluster.yml [hunk 1]
```diff
@@ -1,47 +1,12 @@
...
-     failed_when: runtimedir|length == 0 and site_type != "default"
-
-
   roles:
-    - arc_frontend
-
-
-
+    - role: arc_frontend
+      sessionbasename: '/mnt/grid/sessiondir'
```
Comment (maikenp) [inline playbook_arc_cluster.yml:11]: Not convinced about putting these variables here - as there are many others the user needs to set too. What were your thoughts here?

Comment (sebastian-luna-valero) [reply playbook_arc_cluster.yml:11]: According to my tests in Alma 9 and Debian 11, to get a basic deployment with Slurm + ARC working with Infrastructure Manager I only need to set:
* `sessionbasename`
* `cachebasename`
* and for Debian exclusively: `lrms_bin_path`

This is similar to what we do in the deployment of HTCondor in Infrastructure Manager:
https://github.com/grycap/tosca/blob/main/artifacts/lrms/htcondor_fe_install.yml

How many other variables you think am I missing or will be need to be listed there?

Comment (maikenp) [reply playbook_arc_cluster.yml:11]: I see. 
At least we have to then auto-fill the lrms-type - if we say we just support slurm and htcondor first, this is a simple logic (well, the logic is simple anyway, but would require a long set of if-tests) 
Then I am assuming you are setting the minimal necessary variables to be changed to set up a default version of the ARC-CE. 
And then if the user wants more customisation he is expected to add more of the variables there in the variable list in the playbook - i.e. not touch the defaults/main.yml file? 
I guess that is fine, and is a bit more towards my wish for visibility (where I had the variables in the group_vars/all first. But your suggestion keeps it cleaner and easier integrateable with other playbooks and roles.

Comment (sebastian-luna-valero) [reply playbook_arc_cluster.yml:11]: > And then if the user wants more customisation he is expected to add more of the variables there in the variable list in the playbook - i.e. not touch the defaults/main.yml file?

That's exactly the purpose!

I have just added `slurm` to the examples.

If you agree, please merge so I can proceed with further refactoring.

Thanks!

**Pull request merged**